src/OrchardCore.Modules/OrchardCore.ReCaptcha/Users/Handlers/LoginFormEventEventHandler.cs

```diff
@@ -23,14 +26,12 @@ public Task LoggedInAsync(IUser user)
         return Task.CompletedTask;
     }
 
-    public Task LoggingInAsync(string userName, Action<string, string> reportError)
+    public async Task LoggingInAsync(string userName, Action<string, string> reportError)
     {
-        if (_reCaptchaService.IsThisARobot())
+        if (_reCaptchaService.IsThisARobot() && await _signInManager.GetExternalLoginInfoAsync() == null)
```

Comment:
```suggestion
        if (_reCaptchaService.IsThisARobot() && await _signInManager.GetExternalLoginInfoAsync() is null)
```

Reply:
I still don't think that the pattern matching syntax brings any value when there is no actual pattern to match.